### PR TITLE
[7.x] Replace placeholder for dots and asterisks in validator

### LINKED ISSUE #33357: Validator dotPlaceholder breaks key names returned
- Laravel Version: 7.14.1
- PHP Version: 7.4
- Database Driver & Version: N/a

### Description:
It appears when a request parameter key contains a period "." the Validator.php generates a random string, and replaces the . but does not revert this when returning the parameters via $request->validated() - breaking the expected key name.

If it is required for dots to be converted within the application logic, would expect this to be reverted when returning the parameters from $request->validated();

### Steps To Reproduce:
Create request with rules():

`    public function rules()
    {
        return [
            'example'  => 'bail|required'
       ]
    }
`
Send request with parameters:

$_POST['example] = [ 'a.host.name' => null ];

Output:

$request->all():

`array(
'example' => array('a.host.name' => null)
)`

$request->validated():

`array(
'example' => array('aRANDOMhostRANDOMname' => null)
)`

### PR DESCRIPTION
**Pull request by themsaid**
This PR fixes: https://github.com/laravel/framework/issues/33357

### CODE REVIEW

**Comment by taylorotwell** [comment]
Can we update a test for this?

**Comment by themsaid** [comment]
@taylorotwell test added

File: tests/Validation/ValidationValidatorTest.php [hunk 1]
```diff
@@ -3880,6 +3880,32 @@ public function testPassingSlashVulnerability()
         $this->assertTrue($v->fails());
     }
 
+    public function testPlaceholdersAreReplaced()
```
Comment (klimov-paul) [inline tests/Validation/ValidationValidatorTest.php:3883]: This test does not actually covers the problem described in the related issue: it does not check whether validated data array contains the correct key or not.

Comment (themsaid) [reply tests/Validation/ValidationValidatorTest.php:3883]: Feel free to open a PR with more tests. That's always a good thing :)

Comment (klimov-paul) [reply tests/Validation/ValidationValidatorTest.php:3883]: Already done: #33932

**Pull request merged**